esiclient/v1/node_network.py

```diff
@@ -264,21 +264,48 @@ def take_action(self, parsed_args):
         ironic_client = self.app.client_manager.baremetal
         neutron_client = self.app.client_manager.network
 
-        node = None
-        port = None
+        ports = list((port for port in ironic_client.port.list(detail=True)))
 
+        current_node = None
         with concurrent.futures.ThreadPoolExecutor() as executor:
             f1 = executor.submit(ironic_client.node.get, node_uuid)
+            current_node = f1.result()
+
+        if current_node is not None:
+            node_uuid1 = current_node.uuid
+
+        mapped_node_port_list = []
+        for port in ports:
+            if(port_uuid is not None and port_uuid != ""):
+                if(port.node_uuid == node_uuid1 and port.uuid == port_uuid):
+                    mapped_node_port_list.append(port)
+            else:
+                if(port.node_uuid == node_uuid1):
+                    mapped_node_port_list.append(port)
+        if(len(mapped_node_port_list) == 0):
+            raise exceptions.CommandError(
+                "ERROR: The port is not associated with node {0}".format(
+                    current_node.name))
+        elif(len(mapped_node_port_list) > 1):
+            raise exceptions.CommandError(
+                "ERROR: Multiple ports for {0} Please specify the port".format(
+                    current_node.name))
+        elif(len(mapped_node_port_list) == 1):
+            port_uuid = mapped_node_port_list[0].\
+                internal_info["tenant_vif_port_id"]
+
+        port = None
+
+        with concurrent.futures.ThreadPoolExecutor() as executor:
             f2 = executor.submit(neutron_client.find_port, port_uuid)
-            node = f1.result()
             port = f2.result()
```

Comment:
It looks like you're submitting requests to an executor...but then you're immediately asking for a result. Unless I am missing something, you could replace these lines with:

```suggestion
        port = neutron_client.find_port(port_uuid)
```

And you would have the same behavior.